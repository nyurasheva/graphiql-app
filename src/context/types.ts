export type Translation = {
  [key: string]: string;
};

export type Translations = {
  [key: string]: Translation;
};

export type LocalizationContextType = {
  language: string;
  translations: Translation;
  changeLanguage: (newLanguage: string) => void;
};
