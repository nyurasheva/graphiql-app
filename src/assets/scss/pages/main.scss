/* main pages*/

.title-animation {
  opacity: 0.001;
  transform: translate3d(0, 7rem, 0);

  &.visible {
    opacity: 1;
    transform: translateZ(0);
    transition: transform, opacity;
    transition-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1),
      cubic-bezier(0.42, 0, 1, 1);
    transition-duration: 1s, 0.5s;
    transition-delay: 0s, 0s;
  }
}
