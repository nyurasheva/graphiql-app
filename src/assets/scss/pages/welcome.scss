.welcome {
  @include flexColumn(space-between);
  padding: 0;
  background: $background-dark;

  @include media('xl') {
    padding: 0;
  }

  &__title {
    width: 100%;
    background: $background-gradient;
    padding: 50px 0;

    h1 {
      text-align: left;

      span {
        color: $links;
      }

      @include media('xl') {
        font-size: clamp(14px, 3vw, 28px);
      }
    }

    p {
      text-align: justify;
      margin-top: 10px;
    }
  }

  &__description {
    color: #e7e7e7;
    font-style: italic;

    @include media('xl') {
      display: inline-block;
      width: 100%;
      text-align: center;
    }
  }
}
