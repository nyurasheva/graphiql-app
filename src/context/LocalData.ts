import { Translations } from './types';

// Simulated translations data
export const translationsData: Translations = {
  en: {
    en: 'En',
    ru: '',
    welcome: 'Welcome',
    main: 'Editor',
    registration: 'Registration',
    login: 'Sign in',
    logout: 'Sign out',
    registerTitle: 'Create an account',
    registerSubtitle: 'Registered user benefit:',
    registerText: 'Already have an account? Then just click ',
    loginTitle: 'Sign in',
    loginSubtitle: 'Not registered?',
    loginText: 'Creating an account will take no more than a minute ',
    notfoundText: 'Error',
    notfoundTitle: 'Sorry! We could not find what you were looking for. ',
    notfoundInfo: 'The page was not found.',
    notfoundLink: 'Go to home page',
    submit: 'Submit',
    api: 'API',
    headers: 'Headers',
    variables: 'Variables',
    request: 'Request',
    response: 'Response',
    graphqlTitle: ' — this is the ability to harness all the power',
    graphqlText: 'make a request into your own universe',
    graphqlDescr:
      'The task is to implement a playground/IDE for working with GraphQL queries. GraphQL is a query language used by client applications to handle data. GraphQL is associated with the concept of a schema that allows organizing the creation, reading, updating, and deleting of data in the application. GraphiQL is an open-source tool. In addition to the IDE itself for working with GraphQL queries, authorization capabilities are implemented to grant access to the tool only to authorized users. ',
    dogTitle: `Let's do something together!`,
    dogText1: 'Try to use our editor created thanks to the course ',
    dogText2:
      'By taking courses, you get: free training, technological materials, a mentor, new friends, a certificate and much more.',
    dogBtn: 'Go to the editor',
    aboutTitle: 'You get it!',
    plusTitle: 'Advantages',
    plusBtn: 'Go to the editor',
    mainTitle: 'GraphiQL Editor',
    attentionApi: 'Attention! Submitting API...',
    attentionHeaders: 'Attention! Submitting Headers...',
    attentionVariables: 'Attention! Submitting Variables...',
    docsLink: 'Docs',
    ourTeam: 'Our team',
  },
  ru: {
    en: '',
    ru: 'Рус',
    welcome: 'Приветствие',
    main: 'Редактор',
    registration: 'Регистрация',
    login: 'Вход',
    logout: 'Выход',
    registerTitle: 'Создать учетную запись',
    registerSubtitle: 'Преимущества зарегистрированного пользователя:',
    registerText: 'Уже есть учетная запись? Тогда просто жмите ',
    loginTitle: 'Вход',
    loginSubtitle: 'Не зарегистрированы?',
    loginText: 'Создание учетной записи займет не больше минуты. ',
    notfoundText: 'Ошибка',
    notfoundTitle: 'Извините! Мы не смогли найти то, что вы искали.',
    notfoundInfo: 'Запрашиваемая страница не найдена.',
    notfoundLink: 'Перейти на главную страницу',
    submit: 'Отправить',
    api: 'АПИ',
    headers: 'Заголовок запроса',
    variables: 'Переменные',
    request: 'Запрос',
    response: 'Ответ',
    graphqlTitle: ' — это возможность использовать всю мощь',
    graphqlText: 'сделай запрос в свою собственную вселенную',
    graphqlDescr:
      'Задача состоит в том, чтобы реализовать playground/IDE для работы с запросами GraphQL.GraphQL - это язык запросов, используемый клиентскими приложениями для обработки данных.GraphQL ассоциируется с таким понятием, как schema, которое позволяет организовать создание, чтение, обновление и удаление данных в приложении. GraphiQL - это инструмент с открытым исходным кодом.Помимо самого ide для работы с запросами GraphQL, реализованы возможности авторизации, чтобы предоставить доступ к инструменту только авторизованным пользователям. ',
    dogTitle: 'Давайте делать что-нибудь вместе!',
    dogText1:
      'Попробуйте в использовании наш редактор, созданный благодаря курсу от ',
    dogText2:
      'Обучаясь на курсе вы получаете: бесплатное обучение, учебные материалы, ментора, новых друзей, сертификат и многое другое.',
    dogBtn: 'Перейти в редактор',
    aboutTitle: 'Вы получаете!',
    plusTitle: 'Преимущества',
    plusBtn: 'Перейти в редактор',
    mainTitle: 'GraphiQL Редактор',
    attentionApi: 'Внимание! Отправка API...',
    attentionHeaders: 'Внимание! Отправка заголовков...',
    attentionVariables: 'Внимание! Отправка переменных...',
    docsLink: 'Схема',
    ourTeam: 'Наша команда',
  },
};
