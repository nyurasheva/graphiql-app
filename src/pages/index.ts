export { default as Login } from './Login';
export { default as Welcome } from './Welcome';
export { default as NotFound } from './NotFound';
export { default as Registration } from './Registration';
export { default as Main } from './Main';
