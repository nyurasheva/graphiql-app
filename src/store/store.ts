import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';
import userReducer from './slice/userSlice';
import requestReducer from './slice/requestSlice';

const reducer = combineReducers({
  user: userReducer,
  request: requestReducer,
});

const store = configureStore({
  reducer,
});

setupListeners(store.dispatch);

export default store;
