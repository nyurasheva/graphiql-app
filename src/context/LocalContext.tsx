import React, {
  createContext,
  useState,
  useEffect,
  useContext,
  ReactNode,
} from 'react';
import { LocalizationContextType, Translation } from './types';
import { translationsData } from './LocalData';

const LocalizationContext = createContext<LocalizationContextType | undefined>(
  undefined
);

type LocalizationProviderProps = {
  children: ReactNode;
};

export const LocalizationProvider: React.FC<LocalizationProviderProps> = ({
  children,
}) => {
  const [language, setLanguage] = useState<string>('en');
  const [translations, setTranslations] = useState<Translation>({});

  const changeLanguage = (newLanguage: string) => {
    setLanguage(newLanguage);
  };

  useEffect(() => {
    setTranslations(translationsData[language]);
  }, [language]);

  return (
    <LocalizationContext.Provider
      value={{ language, translations, changeLanguage }}
    >
      {children}
    </LocalizationContext.Provider>
  );
};

export const Localization = () => {
  const context = useContext(LocalizationContext);
  if (!context) {
    throw new Error('Localization must be used within a LocalizationProvider');
  }
  return context;
};
