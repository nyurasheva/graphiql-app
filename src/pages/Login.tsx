import { NavLink, Navigate, useLocation } from 'react-router-dom';
import { MAIN_ROUTE, REGISTRATION_ROUTE } from '../constants/route';
import { LoginForm } from '../components';
import { Localization } from '../context/LocalContext';
import { useAppSelector } from '../store/types';

const Login = () => {
  const { isAuth } = useAppSelector((state) => state.user);
  const { translations } = Localization();
  const location = useLocation();
  if (isAuth) {
    return <Navigate to={MAIN_ROUTE} state={{ from: location }} replace />;
  }
  return (
    <>
      <div className="container">
        <div className="authorization">
          <div className="authorization__container">
            <h1>{translations.loginTitle}</h1>
            <LoginForm />
          </div>
          <div className="authorization__information">
            <h4>{translations.loginSubtitle}</h4>
            <p className="white">{translations.loginText}</p>
            <NavLink to={REGISTRATION_ROUTE}>
              {translations.registerTitle}
            </NavLink>
          </div>
        </div>
      </div>
    </>
  );
};

export default Login;
