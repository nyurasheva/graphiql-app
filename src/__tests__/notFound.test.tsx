import '@testing-library/jest-dom';
import { act, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { LocalizationProvider } from '../context/LocalContext';
import store from '../store/store';
import { NotFound } from '../pages';

describe('EditorSection component', () => {
  it('should be equal EditorSection snapshot', () => {
    let container;

    act(() => {
      ({ container } = render(
        <LocalizationProvider>
          <Provider store={store}>
            <BrowserRouter>
              <NotFound />
            </BrowserRouter>
          </Provider>
        </LocalizationProvider>
      ));
    });

    expect(container).toMatchSnapshot();
  });
});
