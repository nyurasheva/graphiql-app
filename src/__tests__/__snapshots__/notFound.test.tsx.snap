// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorSection component should be equal EditorSection snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="exception"
    >
      <div
        class="exception__code"
      >
        404
        <span
          class="exception__code-txt"
        >
          Error
        </span>
      </div>
      <div
        class="exception__title-info"
      >
        <h1
          class="exception__title"
        >
          Sorry! We could not find what you were looking for. 
        </h1>
        <p
          class="exception__info"
        >
          The page was not found.
        </p>
        <div
          class="exception__links"
        >
          <div
            class="exception__links-item"
          >
            <a
              aria-current="page"
              class="exception__links-a active"
              href="/"
              title=""
            >
              Go to home page
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
